packages/hub/routes/custodial-wallet.ts

```diff
@@ -0,0 +1,82 @@
+import Koa from 'koa';
+import autoBind from 'auto-bind';
+import DatabaseManager from '../services/database-manager';
+import { inject } from '../di/dependency-injection';
+import Wyre from '../services/wyre';
+import { AuthenticationUtils } from '../utils/authentication';
+
+export default class CustodialWalletRoute {
+  authenticationUtils: AuthenticationUtils = inject('authentication-utils', { as: 'authenticationUtils' });
+  databaseManager: DatabaseManager = inject('database-manager', { as: 'databaseManager' });
+  wyre: Wyre = inject('wyre');
+
+  constructor() {
+    autoBind(this);
+  }
+
+  async get(ctx: Koa.Context) {
+    if (!ensureLoggedIn(ctx)) {
+      return;
+    }
+
+    let userAddress = ctx.state.userAddress.toLowerCase();
+    let db = await this.databaseManager.getClient();
+    let result = await db.query('SELECT id, wyre_wallet_id, deposit_address FROM custodial_wallets WHERE id = $1', [
```

Comment:
It is interesting that the `id` is the user's authenticated layer 2 address. My assumption before looking closely at this code is that a field named "id" would a primary key that is either a randomly generated UUID or an auto-incrementing integer. It might be clearer if this field was `user_address` or something like that and designated as the primary key.

Reply:
sure that's fine. an ethereum address is a really great natural primary key, when it makes sense to use it as a primary we we shouldn't shy away from it.